### PR TITLE
Implement subnet "/64#N" notation in Terraform

### LINKED ISSUE #8432: IPv6 support
**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

With the graduation from IPv6 to beta in Kubernetes https://github.com/kubernetes/enhancements/issues/508#issuecomment-574126795
the project has a CI that tests conformance and e2e in a synthetic environment with KIND.

This guarantees a minimum of quality and reduces the risk of regression, however, it needs more user validation. In order to achieve this is important that the lifecycle projects support it allowing users to deploy IPv6 Kubernetes clusters.

**2. Feel free to provide a design supporting your feature request.**

As of today, only 2 cloud providers support IPv6, since AWS is the more mature one and Azure is working on supporting IPv6 with its cluster management tool it's reasonable to implement IPv6 first in AWS and iterate over it.

I have done a sketch of an API implementation for AWS for the cluster-API project
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1370

The setup of the infrastructure, VMs, VPCs, ... is working, the missing part is the configuration of the Kubernetes components.

In order to configure IPv6 in Kubernetes you just have to configure the components with IPv6 addresses, which is not an opt-in feature.

### PR DESCRIPTION
**Pull request by johngmyers**
Closes #8432

### CODE REVIEW

**Comment by johngmyers** [comment]
@rifelpet I'm having to really abuse the `terraformWriter` interface because all the `Literal` constructors take arguments of type `string` instead of `Literal`.

**Comment by johngmyers** [comment]
@rifelpet I'm not quite sure how to address this PR's abuse of the `Literal` interface. It seems `Literal` is going through a lot of complexity in order to use HCL's constructs to generate the output. I'm not sure why this is necessary, especially since it also generates the output itself in order to serialize to JSON for unit tests.

File: upup/pkg/fi/cloudup/awstasks/vpc.go [hunk 1]
```diff
@@ -296,6 +296,10 @@ func (_ *VPC) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *VPC)
 		return err
 	}
 
+	if err := t.AddOutputVariable("vpc_ipv6_cidr_length", terraformWriter.LiteralFunctionExpression("local.vpc_ipv6_cidr_block == null ? null : tonumber", "regex(\".*/(\\\\d+)\", local.vpc_ipv6_cidr_block)[0]")); err != nil {
```
Comment (rifelpet) [inline upup/pkg/fi/cloudup/awstasks/vpc.go:299]:
```suggestion
	if err := t.AddOutputVariable("vpc_ipv6_cidr_length", terraformWriter.LiteralFunctionExpression("local.vpc_ipv6_cidr_length")); err != nil {
```

**Review by rifelpet — COMMENTED** [review]
Looking at [how we interact with hclwrite](https://github.com/kubernetes/kops/blob/55b5e998557e6ba230dc65a24b64db9027416d15/upup/pkg/fi/cloudup/terraform/hcl2.go#L111), we might be able to have the Literal struct just contain a `hclwrite.Tokens` field and have all of the constructors build the tokens themselves. That may allow for more flexibility in those constructors without all the complexities of their string arguments.

File: upup/pkg/fi/cloudup/awstasks/vpc.go (hunk 1 above)
Comment (johngmyers) [reply upup/pkg/fi/cloudup/awstasks/vpc.go:299]: This `AddOutputVariable` call is what defines `local.vpc_ipv6_cidr_length`. You can't define a local variable as being itself.

**Comment by johngmyers** [comment]
I would prefer we refactor the `Literal` constructors as a followup PR.

**Review by rifelpet — APPROVED** [review]

**Pull request merged**